src/main/java/com/codecool/virtualpub/ui/Display.java

```diff
@@ -1,4 +1,119 @@
+        try{
+            System.out.println("\n" + "Customer: " + script.getSentences().get(index) + "\n");
+        } catch (Exception e) {
+            System.out.println("Script not found");
+        }
+    }
+
+    /**
+     * Displays the details of a drink (drink type and brand) to the console.
+     *
+     * @param drink The {@link Drink} object containing the drink details to be displayed.
+     */
+    public void displayDrink(Drink drink) {
+        System.out.println(drink.getDrinkType().getType() + ": " + drink.getBrand());
+    }
+
+    /**
+     * Displays the stock of drinks to the console. The drinks are grouped by {@link DrinkType}.
+     * Each drink type is followed by the available brands, their respective amounts, and the price per unit.
+     *
+     * @param drinks A list of {@link Drink} objects representing the stock of drinks.
+     */
+    public void displayStock(List<Drink> drinks) {
+        Map<DrinkType, StringBuilder> drinkTypeMap = new HashMap<>();
+
+        for (Drink drink : drinks) {
+            StringBuilder stringBuilder = drinkTypeMap.computeIfAbsent(drink.getDrinkType(), drinkType -> new StringBuilder("\n  " + drinkType + ":\n"));
+
+            stringBuilder.append("    ")
+                    .append(drink.getAmount())
+                    .append(" unit of ")
+                    .append(drink.getBrand())
+                    .append(" - price: ")
+                    .append(drink.getPrice())
+                    .append("/unit\n");
+        }
+
+        System.out.println("\nStock: ");
+        for (StringBuilder stringBuilder : drinkTypeMap.values()) {
+            System.out.print(stringBuilder.toString());
+        }
+    }
+
+    /**
+     * Displays a list of actions performed by the bartender. The actions are numbered.
+     *
+     * @param bartenderName The name of the bartender.
+     * @param actions An array of strings representing the bartender's actions.
+     */
+    public void displayBartenderActions(String bartenderName, String[] actions) {
+        System.out.println("\n" + bartenderName + "'s actions: ");
+        for (int i = 1; i <= actions.length; i++) {
+            System.out.println(i + ": " + actions[i - 1]);
+        }
+    }
+
+    /**
+     * Displays the number of customers remaining in the pub and their names if provided.
+     *
+     * @param numberOfCustomers The number of customers in the pub.
+     * @param customerNames     An optional array of customer names. If null, names are not displayed.
+     */
+    public void displayRemainingCustomers(int numberOfCustomers, String[] customerNames) {
+        System.out.println("\nThere are " + numberOfCustomers + " customers in the pub\n");
+        if (customerNames != null) {
+            System.out.println("Customers in line: ");
+            for (int i = 1; i <= customerNames.length; i++) {
+                System.out.println(i + ": " + customerNames[i - 1]);
+            }
+        }
```

Comment:
This is good as it is, but here's a trick to avoid the extra indentation created by the `if` statement:
```java
    public void displayRemainingCustomers(int numberOfCustomers, String[] customerNames) {
        System.out.println("\nThere are " + numberOfCustomers + " customers in the pub\n");
        if (customerNames == null) {
            return;
        }
        System.out.println("Customers in line: ");
        for (int i = 1; i <= customerNames.length; i++) {
           System.out.println(i + ": " + customerNames[i - 1]);
        }
    }
   ```

Or even better, you can remove the `null` check entirely if you move the first line into `displayRemainingCustomers` and just call `displayRemainingCustomers` from this function instead of the other way around.